lib/lint/linter.js

```diff
@@ -0,0 +1,179 @@
+'use babel'
+
+import {CompositeDisposable} from 'atom'
+import path from 'path'
+
+function capitalizeFirstLetter (str) {
+  if (!str) {
+    return str
+  }
+  return str.charAt(0).toUpperCase() + str.slice(1)
+}
+
+class GometalinterLinter {
+  constructor (goconfig) {
+    this.goconfig = goconfig
+    this.subscriptions = new CompositeDisposable()
+
+    this.name = 'gometalinter'
+    this.grammarScopes = ['source.go']
+    this.scope = 'project'
+    this.lintOnFly = false
+  }
+
+  dispose () {
+    if (this.subscriptions) {
+      this.subscriptions.dispose()
+    }
+    this.subscriptions = null
+    this.goconfig = null
+    this.name = null
+    this.grammarScopes = null
+    this.lintOnFly = null
+  }
+
+  ready () {
+    if (!this.goconfig) {
+      return false
+    }
+
+    return true
+  }
+
+  lint (editor) {
+    if (!this.ready() || !editor) {
+      return []
+    }
+
+    const buffer = editor.getBuffer()
+    if (!buffer) {
+      return []
+    }
+    let args = atom.config.get('go-plus.gometalinter.args')
```

Comment:
@lloiser thanks for catching all of these errors

you have push access right?  why not just push fixes when you spot something wrong?  that way we won't forget to go back to any of the things you've pointed out.